.github/workflows/pyrealm_ci.yaml

```diff
@@ -55,22 +55,36 @@ jobs:
       env:
         CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
 
-    - name: Run profiling
+    - name: Run the profiling test suite
       id: profiling
-      if: steps.lfs.outcome == 'success'
+      if: success() && (matrix.os == 'ubuntu-latest' && matrix.python-version == 3.10)
       run: |
         sudo apt-get install graphviz
         poetry run pytest -m "profiling" --profile-svg
-
-    - name: Report performance (this step will fail if performance degrades)
-      if: steps.profiling.outcome == 'success'
-      run: poetry run python profiling/report.py --event ${{ github.event_name }}
-
+        calculatedSha=$(git rev-parse --short ${{ github.sha }})
+        echo "COMMIT_SHORT_SHA=$calculatedSha" >> $GITHUB_ENV
+
+    # This step only runs if the profiling succeeds - so only runs on ubuntu/3.10 and 
+    # will itself fail if the benchmarking fails
+    - name: Benchmark performance
+      id: benchmarking
+      if: ${{steps.profiling.outcome == 'success'}}
+      run: poetry run python profiling/run_benchmarking.py prof/combined.prof profiling/profiling-database.csv profiling/benchmark-fails.csv ${{ env.COMMIT_SHORT_SHA }} --update-on-pass --plot-path profiling/performance-plot.png
+
+    # If the benchmarking succeeds, then update the call graph. This again only runs on
+    # the ubuntu/3.10 runner used for the profiling.
+    - name: Update call graph
+      id: update_call_graph
+      if: ${{steps.benchmarking.outcome == 'success'}}
+      run: cp prof/combined.svg profiling/call-graph.svg
+    
+    # This runs if the benchmark passes or fails (so still only on ubuntu/3.10) and is
+    # skipped on other runners or if the profiling itself fails.
     - uses: stefanzweifel/git-auto-commit-action@v5
-      if: ${{ always() && steps.profiling.outcome != 'skipped' && github.event_name == 'merge_group' }}
+      if: ${{steps.benchmarking.outcome != 'skipped'}}
```

Comment:
If `always()` is removed, this step will not be executed when the benchmark step fails, so I think `benchmark-fails.csv` will not get updated by the auto-commit.

Reply:
Yup - I'd forgotten that `...outcome == 'failed'` will skip the step.